### DISCUSSION TITLE
Can we use the CircuitBreaker in multi-level failover

### DISCUSSION THREAD

**Discussion by vittaliy79** [resilience4j/resilience4j#1798] [Q&A] [answered]
Hi,

I'm trying to find a solution of using the CircuitBreaker in Spring Boot for the failover scenario where failback method with second datacenter (DC) might also fail and we need the **second** failback method to switch to the third or even fourth DC. 
Does CircuitBreaker supports that scenario? I've tried just use the _CircuitBreaker_ annotation on the fallback method and set another method name as the fallbackMethod value but it didn't help. 
```
    @CircuitBreaker(name = "sendPrimaryRequestMessage", fallbackMethod = "sendFallbackRequestMessage")
    public Mono<Response> sendPrimaryRequestMessage(...)

    @CircuitBreaker(name = "sendFallbackRequestMessage", fallbackMethod = "sendSecondFallbackRequestMessage")
    public Mono<Response> sendFallbackRequestMessage(...)

    public Mono<Response> sendSecondFallbackRequestMessage(..)
```

I only see the **sendPrimaryRequestMessage** CircuitBreaker in the metrics.

Could you advice or help? Thank you!
Vitaliy

**Comment by RobWin** [comment, marked as answer]
It does not work because of Spring AOP. 
Annotations don't work if the method `sendFallbackRequestMessage` is called from within the same class. 

It could work with annotations, if you extract the code into a separate service class.

Service A:
```
@CircuitBreaker(name = "sendPrimaryRequestMessage", fallbackMethod = "sendFallbackRequestMessage")
public Mono<Response> sendPrimaryRequestMessage(...)

public Mono<Response> sendFallbackRequestMessage(  
    serviceB.sendFallbackRequestMessage()
)
```

Service  B:
```
    @CircuitBreaker(name = "sendFallbackRequestMessage", fallbackMethod = "sendSecondFallbackRequestMessage")
    public Mono<Response> sendFallbackRequestMessage(...)

    public Mono<Response> sendSecondFallbackRequestMessage(..)

```


But I think it's more simple to don't use annotations, but inject the CircuitBreakerRegistry into your class and
write the code yourself by using try/catch and the CircuitBreaker API directly.
